Assets/Scripts/Environment/BreakableWall.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BreakableWall : MonoBehaviour
+{
+    /* Notes:
+     * The wall object with this script must also have a collider component
+     */
+    public int wallHealth = 10;
+    public int playerCollisionDamage= 5;
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (wallHealth <= 0)
+        {
+            DestroyWall();
+        }
+
+    }
+
+    void DestroyWall()
+    {
+        //Debug.Log("Wall Destroyed");
+        Destroy(gameObject);
+    }
+    public void OnTriggerEnter(Collider other)
+    {
+        if (other.gameObject.CompareTag("Player"))
+        {
+            //Debug.Log("Hit player");
+            //collision.gameObject.GetComponent<PlayerHealth>().TakeDamage(playerCollisionDamage);
+
+            DestroyWall();
+        }
+
+        if (other.gameObject.CompareTag("Bullet"))
```

Comment:
This should be removed from here and be done in the Bullet script OnCollisionEnter method, you usually want the thing applying the damage the call the method takeDamage from the thing it's colliding with, right now the bullet Script doesn't have that functionality so I will take care of the wall taking bullet damage, however this if statement should be removed as it is no longer needed here.